### DISCUSSION TITLE
Is there a way to execute only one onSuccess callback after the network request, no matter how many instances of a certain query there are on a page?

### DISCUSSION THREAD

**Discussion by smashercosmo** [TanStack/query#4859] [Q&A] [answered]
According to this twitter discussion https://twitter.com/shcoding/status/1617206486660161537 when there are multiple instances of a certain query on a page all their onSuccess callbacks will be executed after the network request. Sometimes it can be wasteful. Is there a way to avoid that?

**Comment by TkDodo** [comment, marked as answer]
https://tanstack.com/query/v4/docs/react/reference/QueryCache#global-callbacks

**Reply by smashercosmo** [reply]
Thank you. Another reminder to read the docs thoroughly :)
